otwcode/otwarchive
# AO3-5689 Refactor code to avoid use of eval

# Pull Request Checklist


## Issue

https://otwarchive.atlassian.net/browse/AO3-5689

## Purpose

Refactors the issues brakeman points out for eval

## Testing Instructions

Editing the settings on a prompt meme and gift exchange when there were none originally set
Editing the settings on a collections when you are the owner and edit the settings from the sidebar.
Deleting a collections( prompt and gift ) form the collection profile.
And the same from the challenge deletion form
Check that the user stats are right, check all the things stat wise “"date,"hits"  "kudos": work.kudos.count,      "comment_thread_count":  "bookmarks count":   "subscriptions.count": work.subscriptions.count,      "word_count"”
creating and editing a collection
